install/ui/src/freeipa/topology.js

```diff
@@ -497,23 +497,40 @@ topology.servers_search_facet = function(spec, no_init) {
             on_success(data, text_status, xhr);
 
             var result = data.result.results;
-            var counter = 0;
+            var ca_counter = 0;
+            var kra_counter = 0;
 
             for (var i=0, l=result.length; i<l; i++) {
                 var current = result[i];
                 var roles = current.result.enabled_role_servrole;
                 for (var k=0, m=roles.length; k<m; k++) {
-                    if (roles[k] === 'CA server') counter++;
+                    if (roles[k] === 'CA server') ca_counter++;
+                    if (roles[k] === 'KRA server') kra_counter++;
                 }
             }
 
             // Create dialog and show it only when there is only one CA server
-            if (counter != 1) return;
+            if (ca_counter != 1 && kra_counter != 1) return;
+
+            var messages = [];
+            if (ca_counter == 1)
+                messages.push(
+                {
+                    field: false,
+                    $type: 'html',
+                    html: text.get('@i18n:objects.servers.ca_warning_message')
+                });
+            if (kra_counter == 1)
+                messages.push(
+                {
+                    field: false,
+                    $type: 'html',
+                    html: text.get('@i18n:objects.servers.kra_warning_message')
+                });
 
-            var message = text.get('@i18n:objects.servers.ca_warning_message');
             var dialog = IPA.dialog({
-                name: 'ca_warning',
-                title: '@i18n:objects.servers.ca_warning_title',
+                name: 'dogtag_warning',
```

Comment:
Can you dynamically create that list? something like `Warning: Only One {roles} Server Detected`